src/transformState.js

```diff
@@ -5,7 +5,25 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          state[key] = obj.extraData[key];
+        };
+        break;
+      case 'removeProperties':
+        obj.keysToRemove.forEach(el => delete state[el]);
+        break;
+      default:
```

Comment:
This case should work only when the action is 'clear', but if it is the default it will work in all cases except listed above.
You should leave 'clear' here and add a default case to handle "wrong" inputs